identity/backend/src/main/java/io/camunda/identity/usermanagement/service/GroupService.java

```diff
@@ -47,8 +58,21 @@ public void deleteGroup(final CamundaGroup group) {
     camundaUserDetailsManager.deleteGroup(group.name());
   }
 
-  public CamundaGroup updateGroup(final String name, final CamundaGroup group) {
+  public void deleteGroupById(final Long groupId) {
+    if (!groupRepository.existsById(groupId)) {
+      throw new RuntimeException("group.notFound");
+    }
+    groupRepository.deleteById(groupId);
+  }
+
+  public CamundaGroup renameGroup(final String name, final CamundaGroup group) {
```

Comment:
There should be no need to have multiple methods to rename a group. A simple updateGroup method should also do it:

```java
  public CamundaGroup updateGroup(final CamundaGroup group) {
    final var existingGroup =
        groupRepository.findById(groupId).orElseThrow(() -> new RuntimeException("group.notFound"));
    existingGroup.setName(group.name());
    groupRepository.save(existingGroup);
    return findGroupById(group.id()); // <-- or we can just `return group;`? Without optimistic locking there should be no difference
  }
```

Reply:
I'll remove the unnecessary method, that must be a remainder of before using IDs. Thanks for catching that!

As for the logic of `updateGroup` though, currently we're relying on the `JdbcUserDetailsManager` and do not even have a fully-fledged Group entity but use a record.

I'd keep the logic as is (find group and throw if not exist, update group via `JdbcUserDetailsManager` and return group dto with updated data.

Wdyt?

Reply:
We actually have a group entity: https://github.com/camunda/camunda/blob/4180de9b4f4515134383435618659765565356af/identity/backend/src/main/java/io/camunda/identity/usermanagement/model/Group.java#L17

But as the group only has the name right now, it makes no difference how it is updated. If we need to add additional fields, we can refactor it later

Reply:
You're right ... I need to refactor that naming, it's very confusing with CamundaGroup being the group variable but a record dto and returned by a repo ... 

I'll make it nice quickly, better not to build (even if only small) tech debt here...

Reply:
That's a good decision, thank you!